### PR TITLE
cleanup(js): factor code-frames in nx

### PR DESCRIPTION
**Pull request by vicb**
I recently sent a PR to display nicely formatted JSON errors (https://github.com/nrwl/nx/pull/14293) and then realize that there was something similar for JS in the `js` package - code-frames.

This PR factor the code-frames code in `nx` so that it is used in both cases (formatting JSON and JS errors).

This PR also clean up unused code in `packages/js/src/utils/code-frames/identifiers.ts`. It is very unlikely that someone else use the removed code. To make sure I run  a [query](https://sourcegraph.com/search?q=context:global+code-frames/&patternType=standard&sm=1&groupBy=repo) on github an the only usages were from inside nx.

The only difference between formatting a JS error vs a JSON error is that `codeFrameColumns` use to highlight the JS code passed to it. I have changed that so that the highlighted code is now passed in the options. That is used for JS but not for JSON - we don't highlight JSON code.

The colored test snapshots are not very readable but I made sure all of them are correct in `bash`:

```bash
$ echo  "InvalidCommentToken in JSON at 2:7
      ^[[0m ^[[90m 1 | ^[[39m{^[[0m
      ^[[0m^[[31m^[[1m>^[[22m^[[39m^[[90m 2 | ^[[39m      //\\"test\\": 123,^[[0m
      ^[[0m ^[[90m   | ^[[39m      ^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[31m^[[1m^^[[22m^[[39m^[[0m
      ^[[0m ^[[90m 3 | ^[[39m      \\"nested\\": {^[[0m
      ^[[0m ^[[90m 4 | ^[[39m          \\"test\\": 123^[[0m
      ^[[0m ^[[90m 5 | ^[[39m          /*^[[0m
      "
InvalidCommentToken in JSON at 2:7
        1 | {
      > 2 |       //\test\: 123,
          |       ^^^^^^^^^^^^^^
        3 |       \nested\: {
        4 |           \test\: 123
        5 |           /*     
```

### CODE REVIEW

File: packages/nx/src/utils/code-frames.ts [hunk 1]
```diff
@@ -103,14 +101,18 @@ function getMarkerLines(
...
+    highlightedLines?: string;
+  } = {}
 ): string {
   const defs = getDefs(chalk);
   const lines = rawLines.split(NEWLINE);
   const { start, end, markerLines } = getMarkerLines(loc, lines, opts);
 
   const numberMaxWidth = String(end).length;
-  const highlightedLines = highlight(rawLines);
+  const highlightedLines = opts.highlightedLines ?? rawLines;
```
Comment (meeroslav) [inline packages/nx/src/utils/code-frames.ts:115]: Additionally, I would prefer to keep the old functionality, but have `highlight` be an external callback function, e.g.

```typescript
export function codeFrameColumns(
  rawLines: string,
  loc: NodeLocation,
  opts: {
    linesAbove?: number;
    linesBelow?: number;
    highlight?: (rawLines: string) => string;
  } = {}
): string {
   //...
   const highlightedLines = opts.highlight ? highlight(rawLines) : rawLines;
   //...   
 }
```

The function would be then called:
```typescript
import { highlight } from '../code-frames/highlight';

message = codeFrameColumns(
  code,
  { start: { line, column } },
  { highlight }
);
```

**Review by meeroslav — APPROVED** [review]
Left one comment regarding the highlight function, but it's not a show stopper.

**Comment by jaysoo** [comment]
Are there any concerns for terminals that don't display colors? Maybe some CI systems?

For local development this makes sense, just want to check if we thought about fallbacks.

**Comment by vicb** [comment]
@meeroslav great suggestion. I'll update the PR soonish.

@jaysoo my understanding is that chalk [handles that](https://github.com/chalk/chalk#supportscolor). Also this PR does not add colors but only re-use the existing JS error display for JSON.

Thanks to both of you for the feedback

**Comment by vicb** [comment]
I have rebased/updated the PR. Should be ready to go in if the CI is green.

**Comment by jaysoo** [comment]
Thanks. Let's merge it.

**Comment by vicb** [comment]
Thanks!

**Pull request merged**